Product/Production/Services/AuditRepositoryCore/src/test/java/gov/hhs/fha/nhinc/auditrepository/nhinc/AuditRepositoryOrchImplTest.java

```diff
@@ -0,0 +1,100 @@
+import gov.hhs.fha.nhinc.transform.audit.AuditDataTransformHelper;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import org.junit.Test;
+
+/**
+ *
+ * @author tjafri
+ */
+public class AuditRepositoryOrchImplTest {
+
+    private final String EVENT_TYPE = "TestService";
+    private final String EVENT_ID_CODE = "110112";
+    private final String EVENT_ID_CODE_SYS_NAME = "DCM";
+    private final String EVENT_ID_CODE_DISP_NAME = "Query";
+    private final String DIRECTION = "Outbound";
+    private final String EVENT_ACTION_CODE = "R";
+    private final String USER_NAME = "testUser";
+    private final String RELATES_TO_1 = "val1";
+    private final String RELATES_TO_2 = "val2";
+
+    @Test
+    public void testCreateDBAuditObj() {
+        AuditRepositoryOrchImpl auditRepo = new AuditRepositoryOrchImpl();
+        LogEventSecureRequestType auditObj = createLogEventSecureObj();
+        AssertionType assertion = createAssertion();
+        AuditRepositoryRecord dbRec = auditRepo.createDBAuditObj(auditObj, assertion);
+        assertEquals("AuditRepositoryRecord.Direction mismatch", dbRec.getDirection(), auditObj.getDirection());
+        assertEquals("AuditRepositoryRecord.EventType mismatch", dbRec.getEventType(), EVENT_TYPE);
+        assertEquals("AuditRepositoryRecord.EventId mismatch", dbRec.getEventId(), EVENT_ID_CODE_DISP_NAME);
+        assertEquals("AuditRepositoryRecord.RelatesTo", dbRec.getRelatesTo(), RELATES_TO_1);
+        assertNotNull("AuditRepositoryRecord.MessageId", dbRec.getMessageId());
+    }
+
+    private LogEventSecureRequestType createLogEventSecureObj() {
+        LogEventSecureRequestType logObj = new LogEventSecureRequestType();
+        AuditMessageType audit = createAuditMessageType();
+        logObj.setEventType(EVENT_TYPE);
+        logObj.setDirection(DIRECTION);
+        logObj.setAuditMessage(audit);
+        logObj.setEventID(audit.getEventIdentification().getEventID().getDisplayName());
+        logObj.setEventOutcomeIndicator(audit.getEventIdentification().getEventOutcomeIndicator());
+        return logObj;
+    }
+
+    private AuditMessageType createAuditMessageType() {
+        AuditMessageType auditObj = new AuditMessageType();
+        auditObj.setEventIdentification(createEventIdentification());
+        return auditObj;
+    }
+
+    private EventIdentificationType createEventIdentification() {
+        return AuditDataTransformHelper.createEventIdentification(EVENT_ACTION_CODE, 0,
+            AuditDataTransformHelper.createEventId(EVENT_ID_CODE, null, EVENT_ID_CODE_SYS_NAME, EVENT_ID_CODE_DISP_NAME));
+    }
+
+    private AssertionType createAssertion() {
+        AssertionType assertion = new AssertionType();
+        UserType user = new UserType();
+        user.setUserName(USER_NAME);
+        assertion.setUserInfo(user);
+        assertion.getRelatesToList().add(RELATES_TO_1);
+        assertion.getRelatesToList().add(RELATES_TO_2);
```

Comment:
We relatesTo is a List in AssertionType object. We discussed that for logging will pick the first one. Just trying to check in this test case  that even if we have 2 relatesTo we will only pick the first one.